ale_linters/haskell/ghc_mod.vim

```diff
@@ -0,0 +1,18 @@
+" Author: wizzup <wizzup@gmail.com>
+" Description: ghc-mod for Haskell files
+
+call ale#Set('haskell_ghc_mod_executable', 'ghc-mod')
+
+function! ale_linters#haskell#ghc_mod#GetCommand (buffer) abort
+    let l:executable = ale#Var(a:buffer, 'haskell_ghc_mod_executable')
+    return ale#handlers#haskell_stack#EscapeExecutable(l:executable, 'ghc-mod')
+\       . ' --map-file %s=%t check %s'
+endfunction
+
+call ale#linter#Define('haskell', {
+\   'name': 'ghc_mod',
+\   'aliases': ['ghc-mod'],
+\   'executable': 'ghc-mod',
```

Comment:
You should add `executable_callback` here and you can use `ale#VarFunc(''haskell_ghc_mod_executable')` for that. Then ALE will check if the stack executable is executable before running the linter, instead of just `ghc-mod`.

Reply:
So with this method the `%e` is not viable in the command?

Reply:
The command you have is fine, you just need to tell ALE what the executable is too, so it can check if that's executable.